warp-ds/drive
# feat: purge component classes

A part of https://nmp-jira.atlassian.net/jira/software/projects/WARP/boards/15?label=web&selectedIssue=WARP-302

As we are publishing the component.css to EIK our consumers do not need to safelist the classes anymore. They can use the css directly from EIK. Benefits: cached css instead of every app safelists the classes. Could be added to the template etc

And as we don't want duplicated styles to be generated we can finally use externalizedClasses and postprocess feature. We will ALWAYS purge classes so it is crucial for our consumers to use `warp.css` somewhere in their setup.

- [x] would like to skip fetching if `development: true`

????? consumers will need to add an async wrapper :(
 
In the uno.config 👇 

Before|After
---|---
|<pre><code>export default (() => {</code><br>  <code>return defineConfig({</code><br><code>    presets: [presetWarp()],  </code><br><code>  });</code><br><code>})();</code></pre> | <pre><code>export default (async () => {</code><br><code>  const asyncPreset = await presetWarp()</code><br><code>  return defineConfig({</code><br><code>    presets: [asyncPreset]</code><br><code>  })</code><br><code>})();</code></pre>|

**TO TEST**
This PR is needed to be tested together with https://github.com/warp-ds/css/pull/56
- run build in both packages and link them both in the test application you want to test in, for example any of the component frameworks repos (YOU'LL NEED TO REMOVE SAFELIST FROM THE CONFIG AND ADD `warp.css` where it is applicable)
- observe that everything looks like before
- add e.g. `p-16` classes to any element
- observer `p-16` in the generated uno.css
- change `development` to false and rebuild
- observe that `p-16` is no longer in the uno-css
